server/adaptors/integrations/integrations_adaptor.ts

```diff
@@ -0,0 +1,32 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+export interface IntegrationsAdaptor {
+  getIntegrationTemplates: (
+    query?: IntegrationTemplateQuery
+  ) => Promise<IntegrationTemplateSearchResult>;
+
+  getIntegrationInstances: (
+    query?: IntegrationInstanceQuery
+  ) => Promise<IntegrationInstancesSearchResult>;
+
+  getIntegrationInstance: (query?: IntegrationInstanceQuery) => Promise<IntegrationInstanceResult>;
+
+  loadIntegrationInstance: (
+    templateName: string,
+    name: string,
+    dataSource: string
+  ) => Promise<IntegrationInstance>;
+
+  deleteIntegrationInstance: (id: string) => Promise<any>;
+
+  getStatic: (templateName: string, path: string) => Promise<Buffer>;
+
+  getSchemas: (
+    templateName: string
+  ) => Promise<{ mappings: { [key: string]: any }; schemas: { [key: string]: any } }>;
```

Comment:
np. if the type is dynamic then use `unknown` instead of `any`? similarly above if you don't need the return value of delete calls, use `void`?

also here does it actually allow arbitrary types to be in mappings and schemas?